presto-main/src/test/java/com/facebook/presto/cost/TestStatisticRange.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.cost.EstimateAssertion.assertEstimateEquals;
+import static java.lang.Double.NEGATIVE_INFINITY;
+import static java.lang.Double.POSITIVE_INFINITY;
+
+public class TestStatisticRange
+{
+    @Test
+    public void testOverlapPercentWith()
+    {
+        StatisticRange zeroToTen = range(0, 10, 10);
+        StatisticRange empty = StatisticRange.empty();
+
+        // Equal ranges
+        assertOverlap(zeroToTen, range(0, 10, 5), 1);
+        assertOverlap(zeroToTen, range(0, 10, 20), 1);
+        assertOverlap(zeroToTen, range(0, 10, 20), 1);
+
+        // Some overlap
+        assertOverlap(zeroToTen, range(5, 3000, 3), 0.5);
```

Comment:
As I understand, overlapPercent do not take consider `distinctValueCount`, right?

Reply:
In case of ranges with known ends -- yes. If we don't know the range ends (no stats or this is `varchar`), the NDV matters.